### PR TITLE
Silex 2

### LINKED ISSUE #8: Support Silex 2
Hi!

Do you have plans to support Silex 2?

Thanks!

### PR DESCRIPTION
**Pull request by hkdobrev**
Add Silex 2 support.
Closes #8.

### CODE REVIEW

**Comment by asiviero** [comment]
wouldn't this break backwards compatibility?

**Comment by hkdobrev** [comment]
@asiviero Definitely! This should be released as a new minor version `0.2.0` (according to semver rules when the major version is still `0`) or you could release it as `1.0.0`.

However, this would allow users of Silex 2 which has been out for some time to use the provider.

File: src/TrailingSlashControllerProvider.php [hunk 1]
```diff
@@ -41,18 +42,18 @@ public function connect(Application $app)
 
             $request = Request::create(
                 '/' . $resource . '/',
-                $app['request']->getMethod(),
+                $app['request_stack']->getCurrentRequest()->getMethod(),
```
Comment (h-bragg) [inline src/TrailingSlashControllerProvider.php]: Looks like it is worth extracting `$app['request_stack']->getCurrentRequest()` into a variable now

**Review by h-bragg — CHANGES_REQUESTED** [review]
Hi! 

Thanks for the PR! Excellent stuff. 

I expect we would bump this to 1.0 just to be clear.

the `--prefer-lowest --prefer-stable` test on travis is failing. 

you can run:

```shell
make 'composer-update --prefer-lowest --prefer-stable'
make test-unit
```

locally to test. 

I expect:

```diff
     "require-dev": {
          "mockery/mockery": "^0.9.4",
 -        "phpunit/phpunit": "^5.0",
 +        "phpunit/phpunit": "^5.1.1",
          "roave/security-advisories": "dev-master"
      },
```

would fix the issue.

Or I have created a PR: #10 that should help.

**Comment by asiviero** [comment]
@hkdobrev sure, a version bump might be in order, just wanted to highlight it

**Comment by hkdobrev** [comment]
@asiviero @h-bragg Changes made!

The tests were passing locally for me even when I run the docker container and install the lowest deps.

BTW if you want to continue supporting Silex 1 with bugfixes, you might want to release 1.0 for an earlier commit, release 2.0 with this and then have a branch for 1.x.

**Comment by hkdobrev** [comment]
Tests on Travis CI are still failing 😞 Are they failing for you locally?

**Comment by h-bragg** [comment]
Running this locally I get:

```
[10:07] ~/git/silex-trailing-slash-handler ● make 'composer-update --prefer-stable --prefer-dist --prefer-lowest'
...
[10:13] ~/git/silex-trailing-slash-handler ● make test
docker run --rm -it -v $(pwd):/opt/graze/silex-trailing-slash-handler graze/silex-trailing-slash-handler \
	composer test --ansi
Do not run Composer as root/super user! See https://getcomposer.org/root for details
> vendor/bin/phpunit --colors=always tests/
PHPUnit 5.1.1 by Sebastian Bergmann and contributors.

...............................
Fatal error: Call to undefined method Symfony\Component\Debug\ExceptionHandler::getHtml() in /opt/graze/silex-trailing-slash-handler/vendor/silex/silex/src/Silex/ExceptionHandler.php on line 44
Script vendor/bin/phpunit --colors=always tests/ handling the test event returned with error code 255
make: *** [test-unit] Error 255
```

I guess something else needs updating to a minimum version

**Comment by h-bragg** [comment]
I have updated the master branch with my PR to improve travis and the Makefile.

Might be a few changes to rebase in, but hopefully should help with testing.

`make test-matrix-lowest`

will then test with the lowest allowed dependency versions.

**Comment by h-bragg** [comment]
Looks like the problem is that it uses `symfony/debug` version 2.8. But it only explicitly requires `http-kernel` (for non dev) which requires `debug >= 2.6`.

the `->getHtml` was added in `2.8` which is in their `require-dev` group.

Adding `        "symfony/debug": "~2.8|^3.0"` into `require-dev` makes it all work again.

**Comment by hkdobrev** [comment]
@h-bragg Thanks for the insight! Updated.

**Review by h-bragg — DISMISSED** [review]

**Comment by h-bragg** [comment]
Released the current master to v1. And created a `1.x` branch for future development.

This has changed another file, added a test.

Will merge this and release v2 with when it is rebased.

Cheers for the PR and fixes!

**Comment by hkdobrev** [comment]
Done.

**Comment by h-bragg** [comment]
Looks like the logging test has broken this. Does it not call `url_matcher` in silex 2 any more?

**Comment by hkdobrev** [comment]
I'd look into it. I had some conflicts around that.

**Comment by h-bragg** [comment]
Looks like they renamed `url_matcher` to `request_matcher`.

The following patch fixes this and adds a test to ensure it 404's for `/` routes instead of `301` which is default (to ensure the request_matcher is being overridden).

[edit, i created a PR into your PR for this instead]

**Review by h-bragg — APPROVED** [review]

**Comment by hkdobrev** [comment]
@h-bragg Thanks for the insights and help! 🙌

**Pull request merged**